cobbler/items/item.py

```diff
@@ -151,6 +310,25 @@ def __eq__(self, other):
             return self._uid == other.uid
         return False
 
+    def __hash__(self):
+        """
+        Hash table for Items.
+
+        :return: hash(uid).
+        """
+        return hash(self._uid)
```

Comment:
That will probably already be a huge speedup. Do you see any other dunder methods that are maybe useful to have on our items?

Reply:
I wasn't looking for specifically useful methods to include in the code, but I'll think about it. Hash I needed so that the  code with sets could work:
```python
cobbler/items/item.py:
        results = set(childs)
        childs.append(self)
        for child in childs:
            for item_type in Item.TYPE_DEPENDENCIES[child.COLLECTION_TYPE]:
                if item_type is not None and len(item_type) == 2:
                    dep_type_items = self.api.find_items(
                        item_type[0], {item_type[1]: child.name}
                    )
                    results.update(dep_type_items)
                    for dep_item in dep_type_items:
                        results.update(dep_item.descendants)
        return list(results)
```

Reply:
If you think about it that is enough for me atm. Resolving discussion for now.